### PR TITLE
Fix the path to Dockerfile.

### PR DESCRIPTION
**Pull request by ravimeda**
Path to Dockerfile was the containing directory and not the the file. Update the path to include the filename.

https://github.com/dotnet/core-setup/issues/1757

I'll send similar PRs for the two servicing branches.

### CODE REVIEW

**Comment by MichaelSimons** [comment]
@ravimeda, Can you explain what steps you have taken to test this locally?

**Comment by ravimeda** [comment]
Performed the following in a terminal window:

1. DOCKERFILE="\repos\core-setup\scripts\docker\centos.7".  Note there is no slash at end.
2. Run `image=$(grep -i "^FROM " "$DOCKERFILE" | awk '{ print $2 }')`
3. Above step fails saying path is a directory.
4. Run `image=$(grep -i "^FROM " "$DOCKERFILE/Dockerfile" | awk '{ print $2 }')`
5. Command succeeds. `image` is set to `centos:7.1.1503`

Please let me know if other tests are needed.

**Comment by ravimeda** [comment]
For context, in testing the change in PR #1747, I set `DOCKERFILE` to the file and not the directory.  Given the name of the variable, I assumed it pointed to the file. This makes me wonder if we should rename the variable accordingly.

File: scripts/dockerrun.sh [hunk 1]
```diff
@@ -115,7 +115,7 @@ execute() {
 echo "Building Docker Container using Dockerfile: $DOCKERFILE"
 
 # Get the name of Docker image.
-image=$(grep -i "^FROM " "$DOCKERFILE" | awk '{ print $2 }')
+image=$(grep -i "^FROM " "$DOCKERFILE/Dockerfile" | awk '{ print $2 }')
```
Comment (dagood) [inline scripts/dockerrun.sh]: I think we probably want to support both scenarios: `DOCKERFILE` being a path to the dir, as well as a direct path to the Dockerfile. As far as I can tell, the rest of the scripts work fine with either, and I don't think we should enforce the `Dockerfile` name convention.

I think a solution would be to check for `$DOCKERFILE` being a dir, and if it is, try `$DOCKERFILE/Dockerfile`.

**Comment by MichaelSimons** [comment]
@ravimeda, I think it would be best to test all of your changes as they exist in dockerrun.sh.  To do this I think you should be invoking the script itself.  In this particular case you may decide to comment out the docker run section for testing purposes.  You should also consider how to exercise all code paths such as the retry logic.  You can easily do this by disabling your network or passing in a non-existent docker image.  The bottom line is that we need to make certain our changes are addressing the target problem and adding stability to the build system.

**Comment by ravimeda** [comment]
Thanks @MichaelSimons Tested 'dockerrun.sh' in following cases:

1. No arguments
2. Specified -i, -d or both.  Valid and invalid values.
3. Retry logic for image pulling by turning off internet

Updated the description of Dockerfile argument to indicate that it is path to the folder containing the file.

File: scripts/dockerrun.sh (hunk 1 above)
Comment (ravimeda) [inline scripts/dockerrun.sh]: `DOCKERFILE` should point to the directory containing 'Dockerfile'. Else the following command will fail.  
`docker build --build-arg USER_ID=$(id -u) -t $DOTNET_BUILD_CONTAINER_TAG $DOCKERFILE`
If we want to pass a file then `-t` should be replaced with `-f`

File: scripts/dockerrun.sh [hunk 2]
```diff
@@ -37,7 +37,7 @@ while [[ $# > 0 ]]; do
             echo "Usage: $0 [-d|--dockerfile <Dockerfile>] [-i|--image <ImageName>] <Command>"
             echo ""
             echo "Options:"
-            echo "  <Dockerfile>    The path to the Dockerfile to use to create the build container"
+            echo "  <Dockerfile>    The path to the folder that contains Dockerfile to use to create the build container"
```
Comment (dagood) [inline scripts/dockerrun.sh]: \<nit> "contains **a** Dockerfile"

File: scripts/dockerrun.sh (hunk 1 above)
Comment (MichaelSimons) [inline scripts/dockerrun.sh]: @ravimeda - FYI if we want to pass a file then we would need to add the `-f` option and specify the `Name of the Dockerfile (Default is 'PATH/Dockerfile')`  The `-t` option is the `Name and optionally a tag in the 'name:tag' format (default [])`.  That has to remain as it is.

**Comment by ravimeda** [comment]
Thanks!

**Comment by ravimeda** [comment]
https://github.com/dotnet/core-eng/issues/561

**Pull request merged**